### PR TITLE
 [Dialogs] Fixed retain cycle in MDCDialogTransition

### PR DESCRIPTION
**Pull request by randallli**
Fixed retain cycle in MDCDialogTransition that causes MDCAlertController to never be released and presenting view to never be released.

### CODE REVIEW

**Comment by jverkoey** [comment]
Branch looks like it needs to be rebased

**Review by jverkoey — APPROVED** [review]
Changes LGTM, but make sure the branch is clean before merging.

File: components/Dialogs/src/MDCDialogTransition.m
Comment (ianegordon) [inline components/Dialogs/src/MDCDialogTransition.m:48]: How is this block different?  Is the __weak annotation above sufficient for our needs?

Previously:
_presentationController = XYZ;

With this change:
UIPresentationController *localPresentationController = XYZ;
_presentationController = localPresentationController;

Comment (romoore) [reply components/Dialogs/src/MDCDialogTransition.m:48]: I think I understand the need for a local variable.

Now that the ivar is weak, if it's the only one holding the reference to the newly-allocated `MDCDialogPresentationController`, it will immediately be deallocated. A strongly-retained local variable is required to get the instance back to the caller in the return statement.

Comment (ianegordon) [reply components/Dialogs/src/MDCDialogTransition.m:48]: Ah, we are returning presentationController, not _presentationController.
Disregard.

**Comment by romoore** [comment]
Probably needs a rebase. The failure on Travis was fixed in #2686

**Comment by ianegordon** [comment]
I'm not super familiar with the MM transitions yet, so I'll leave this to @jverkoey .

**Comment by ianegordon** [comment]
Having the author as a-googler is going to cause problems with CLA bot when we cut the release as well.  We may want to avoid that in the future.

**Pull request merged**
